front-end/src/features/productsInCartSlice.js

```diff
@@ -5,72 +5,82 @@ export const productsInCartSlice = createSlice({
   initialState: [],
   reducers: {
     addProductToCart: (state, action) => {
-      if (!action.payload.inStock) {
+      const { id, inStock, selectedAttr } = action.payload;
+
+      if (!inStock) {
         return [...state];
       }
-      const id = action.payload.id;
-      const selectedAttr = action.payload.selectedAttr;
-      const idx = state.findIndex(
+
+      const productIndex = state.findIndex(
```

Comment:
Descriptive naming is nooit 'n slegte ding nie. Ek weet `productIndex` is baie langer as `idx`, maar die cognitive load is baie minder want mens weet presies wat dit is deur die `var` naam te lees.

Wees eerder meer explicit as jy goed name as vague net omdat dit korter is.